server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -2780,22 +2787,23 @@ protected final void refreshIfNeeded(String source, long requestingSeqNo) {
 
     private final class LastRefreshedCheckpointListener implements ReferenceManager.RefreshListener {
         final AtomicLong refreshedCheckpoint;
-        private long pendingCheckpoint;
+        final AtomicLong pendingCheckpoint;
```

Comment:
If all you do is get and set (and don't use any of the check-and-set or atomic increment/decrement features), then I think a volatile primitive long is sufficient.

Reply:
Sure. Moved to volatile.